noir-projects/noir-protocol-circuits/crates/types/src/meta/mod.nr

```diff
@@ -24,204 +24,28 @@ comptime fn get_trait_impl_method(
-        let str_len = length_type.as_constant().unwrap();
-        let mut byte_list = &[];
-
-        // Iterate over each character in the string
-        for _ in 0..str_len {
-            // Recursively call `generate_deserialize_from_fields` for each character in the string
-            let (deserialized_field, num_consumed_in_recursion) = generate_deserialize_from_fields(
-                name,
-                quote {u8}.as_type(),
-                field_array_name,
-                consumed_counter + num_already_consumed,
-                should_unpack,
-            );
-
-            // We should consume just one field in the recursion so we sanity check that
-            assert_eq(
-                num_consumed_in_recursion,
-                1,
-                "Incorrect number of fields consumed in string deserialization",
-            );
-
-            // We increment the consumed counter by 1 as we have consumed one field
-            consumed_counter += 1;
-
-            // We add the deserialized field to the list of deserialized fields.
-            // E.g. `fields[6] as u8`
-            byte_list = byte_list.push_back(deserialized_field);
-        }
-
-        // We construct the string from the deserialized fields
-        let bytes = byte_list.join(quote {,});
-        result = quote { [ $bytes ].as_str_unchecked() };
-    } else {
-        panic(
-            f"Unsupported type for serialization of argument {name} and type {typ}",
-        )
-    }
+/// Returns the number of fields required to pack a type `typ` that implements the `Packable` trait.
+pub comptime fn get_packable_len(typ: Type) -> u32 {
+    // We currently don't seem to have a way to access an associated constant of a `TraitImpl` at comptime so
+    // I work around this by:
+
+    // 1. Getting the packable constraint
+    let packable_constraint: TraitConstraint =
+        quote { crate::traits::Packable }.as_trait_constraint();
+    // 2. Getting the trait impl
+    let trait_impl =
+        typ.get_trait_impl(packable_constraint).expect(f"Type {typ} does not implement Packable");
+    // 3. Getting the return type of the pack method
+    let pack_method_return_type = trait_impl.methods()[0].return_type();
+    // 3. Representing the return type as an array - this gives me (Type, Type) where the first type is the type of
+    // elements in the array and the second is the length of the array.
+    let pack_method_return_type_as_array = pack_method_return_type.as_array().unwrap();
+    // 4. Getting the length and representing it as a constant.
+    let packed_len_as_constant = pack_method_return_type_as_array.1.as_constant().expect(
+        f"Could not represent packed length as a constant when getting packable len for {typ}",
+    );
 
-    (result, consumed_counter)
+    packed_len_as_constant
 }
 
 // Note: The main usecase of this function is gone (it's no longer used in derive_serialize) so there is most likely
```

Comment:
I will drop this function in a followup cleanup PR as its now used only in 1 place and only to get serialized length. Hence it will be replaced with something like `get_serialized_len` that will probably work basically the same as `get_packable_len` above.